xls/contrib/integrator/area_model/area_estimator.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef XLS_AREA_MODEL_AREA_ESTIMATOR_H_
+#define XLS_AREA_MODEL_AREA_ESTIMATOR_H_
+
+#include "absl/status/statusor.h"
+#include "xls/delay_model/delay_estimator.h"
+#include "xls/ir/node.h"
+
+namespace xls {
+
+// Abstraction describing an area model for XLS operations.
+class AreaEstimator {
+ public:
+  AreaEstimator(const DelayEstimator* delay_estimator) : delay_estimator_(delay_estimator) {}
+
+  AreaEstimator(const AreaEstimator& other) = delete;
+  void operator=(const AreaEstimator& other) = delete;
+
+  // Returns the estimated area of the given node. Units of area
+  // depend on the model / data used.
+  // TODO(jbaileyhandle): const node?
```

Comment:
It's fine as non-const:

https://github.com/google/xls/blob/c6672aec22a13a5761a8cd29968bf317177b95b7/xls/dslx/cpp_pos.cc#L41

C++ constness though useful at times can be an awkward fit.

Reply:
Comment removed.